File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int> found, visited;
        int ans = 0;
        for(int i = 0; i< nums.size() ;i++){
           found[nums[i]] = 1;
        }
        for(int i = 0; i < nums.size();i++){
            int cur = 1, curElm = nums[i]-1;
            while(! visited[curElm]&& found[curElm]){
                visited[curElm]=1;
                curElm -- ;
                cur ++;
            }
            curElm = nums[i]+1;
             while(!visited[curElm]&&found[curElm]){
                visited[curElm]=1;
                curElm ++;
                cur ++;
            }
            ans = max(ans, cur);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.cs
public class Solution
{
    public List<List<int>> ThreeSum(int[] nums)
    {
        List<List<int>> ans = new List<List<int>>();
        Array.Sort(nums);
        for (int i = 0; i < nums.Length-2; i++)
        {
            int l = i + 1 , r = nums.Length-1;
            if (i>0&&nums[i] == nums[i - 1])
                continue;
            while (l < r)
            {
                if (nums[l] + nums[r] > -nums[i]) r--;
                else if (nums[l] + nums[r] < -nums[i]) l++;
                else
                {
                    ans.Add(new List<int> { nums[i], nums[l], nums[r] });
                    l++;
                    r--;

                    while (l < r && nums[l] == nums[l - 1])
                        l++;

                    while (l < r && nums[r] == nums[r + 1])
                        r--;
                }

            }
        }
        return ans;
    }
}